prime-router/src/main/kotlin/azure/observability/event/ReportRouteEvent.kt

```diff
@@ -0,0 +1,17 @@
+package gov.cdc.prime.router.azure.observability.event
+
+import gov.cdc.prime.router.ReportId
+import gov.cdc.prime.router.Topic
+
+/**
+ * Event definition for when a report gets routed to a receiver
+ */
+data class ReportRouteEvent(
```

Comment:
The actual important part of this PR!

Do we need any additional fields here for better future-proofing?

Reply:
@arnejduranovic @brick-green @mkalish 

Adding in a bunch of additional fields for future use is fine with me. The only potential issue I see us running into is the size limits of what AppInsights can injest.

https://github.com/MicrosoftDocs/azure-docs/blob/main/includes/application-insights-limits.md
Max size of telemetry item: 64 KB
key length: 150
value length: 8192

If we put as many fields at the top level as possible it shouldn't be a problem but just dropping in an object that serializes to json could potentially blow up if the object is really large. We would have to run some tests with common reports to see what kind of sizes we would be working with.

Reply:
If we serialize an object to JSON, would be still be able to query based on the keys in it? I guess I assumed we could but I'm not sure

Reply:
Also, this ticket doesn't need to add all the fields, but it should set up the architecture so we have the ability to add more very easily in the future without having to refactor.

Reply:
> If we serialize an object to JSON, would be still be able to query based on the keys in it? I guess I assumed we could but I'm not sure. Seems like something we should verify as well.

We can query it but there needs to be an additional step in the query to parse that json. It could slow down the overall query so my recommendation is to keep fields we expect to query a lot at the top level.

Reply:
I would like to see what format this message is (hl7 vs FHIR) and then also the value of field MSH.21. When we add this, would we just add more params here? That doesn't seem maintainable to me. Sounds like we may want to use a builder pattern here so we don't have to change the constructor all the time? https://www.baeldung.com/kotlin/builder-pattern